src/nccl_ofi_scheduler.c

```diff
@@ -22,6 +22,40 @@ static inline size_t sizeof_schedule(int num_rails)
 		+ num_rails * sizeof(nccl_net_ofi_xfer_info_t);
 }
 
+/*
+ * @brief  This function calculates the optimal number of stripes
+ * for the payload size based on the min_stripe_size.
+ * It first determines the initial number of stripes based on the message size
+ * and a minimum stripe size, ensuring that the result is at least 1 and at
+ * most the number of available rails. The function then adjusts the number of
+ * stripes to be the nearest lower factor of the number of rails.
+ *
+ * @param	size
+ * 		The size of the message being transmitted.
+ * @param 	scheduler_p
+ * 		Pointer to threshold scheduler
+ * @param 	num_rails
+ * 		The number of available rails for transmission.
+ *
+ * @return	Returns the adjusted number of stripes.
+ *
+ */
+static int get_num_stripes(nccl_net_ofi_threshold_scheduler_t *scheduler_p, size_t size, int num_rails)
+{
+	/* Number of stripes is atleast 1 for zero-sized messages and at most equal to num of rails */
+	int num_stripes =
+		(int)NCCL_OFI_MAX(1, NCCL_OFI_MIN(NCCL_OFI_DIV_CEIL(size, scheduler_p->min_stripe_size), (unsigned)num_rails));
+
+	/* Ensure num_stripes is the nearest lower factor of the number of rails */
+	for (int i = num_stripes; i >= 0; i--) {
```

Comment:
You'll need a comment to explain, but this should be:

```
for (int i = num_stripes ; i > 1 ; i--)
```

if `num_stripes` is 1, then there's no need to run the check.  `num_rails % 1` will always be true, regardless of the value of `num_rails`.  And when `num_stripes` is 1, you're more likely in a latency sensitive regime, in which case the mod operation will increase latency.  So don't do things that increase latency and aren't necessary.

Reply:
Thanks for pointing this and providing the detailed explanation. I have corrected this.